Add gcc into rhel images

marusak opened the pull request:
* [x] image-refresh rhel-9-0
 * [x] image-refresh rhel-8-5

subscription-manager needs gcc to be able to build itself. Rhel-8 images
have gcc present, but it is missing for rhel-9.

see https://logs.cockpit-project.org/logs/pull-2385-20210906-113027-1c37ba32-rhel-9-0-candlepin-subscription-manager/log.html

cockpituous commented:
image-refresh in progress on 1-cockpit-7.
Log: https://logs.cockpit-project.org/logs/image-refresh-2397-20210909-061212/

martinpitt commented:
Did you verify with a locally updated image that just `gcc` is enough? No glibc-devel, binutils, etc?

cockpituous commented:
image-refresh rhel-9-0 done: https://github.com/cockpit-project/bots/commits/image-refresh-rhel-9-0-20210909-062727

cockpituous commented:
image-refresh in progress on 2-ci-srv-04.
Log: https://logs.cockpit-project.org/logs/image-refresh-2397-20210909-062727/

cockpituous commented:
image-refresh rhel-8-5 done: https://github.com/cockpit-project/bots/commits/image-refresh-rhel-8-5-20210909-064747

marusak commented:
> Did you verify with a locally updated image that just gcc is enough? No glibc-devel, binutils, etc?

No I did not :see_no_evil: Let's see :crossed_fingers:

marusak commented:
For now triggering only `rhel-9-0@candlepin/subscription-manager`. If that builds, then I'll trigger all rhel-9 and 8-5 contexts.

marusak commented:
Tests are running. They fail the same way 8-5 is failing, this needs to be debugged, but definitelly not related to this refresh. Triggering all other related statuses.

marusak commented:
`rhel-8-5@osbuild/cockpit-composer` fails on `testUserAccount`. This test has been busted for a while, right @jkozol ?
`rhel-9-0@candlepin/subscription-manage` fails, this is being fixed in https://github.com/candlepin/subscription-manager/pull/2778 but for that to be able to build, this PR needs to land first.

martinpitt reviewed (approved):
Thanks! Feel free to start a sub-man PR run against this PR, but I figure it wouldn't block us from landing this anyway (even if more packages are missing).

martinpitt commented on images/scripts/rhel.setup:
```diff
@@ -107,6 +107,7 @@ echo foobar | passwd --stdin root
 COCKPIT_DEPS="\
 device-mapper-multipath \
 firewalld \
+gcc \
```
FWIW, this still feels wrong/hackish to me. Packages should be built inside mock, not the main VM image. And sub-man's test/vm.install already installs a few extra packages, so it could just as well install gcc and any other stuff that it needs.

ptoscano replied:
Ah, sure, PRs are welcome ;) Is gcc the only package missing?

marusak commented:
ehm, what?
```
Removed:

Added:
  efibootmgr (16-1.el8)

Changed:
  device-mapper (8:1.02.177-6.el8 -> 8:1.02.177-8.el8)
  device-mapper-event (8:1.02.177-6.el8 -> 8:1.02.177-8.el8)
  device-mapper-event-libs (8:1.02.177-6.el8 -> 8:1.02.177-8.el8)
  device-mapper-libs (8:1.02.177-6.el8 -> 8:1.02.177-8.el8)
  kernel (4.18.0-339.el8 -> 4.18.0-340.el8)
  kernel-core (4.18.0-339.el8 -> 4.18.0-340.el8)
  kernel-headers (4.18.0-339.el8 -> 4.18.0-340.el8)
  kernel-modules (4.18.0-339.el8 -> 4.18.0-340.el8)
  kernel-tools (4.18.0-339.el8 -> 4.18.0-340.el8)
  kernel-tools-libs (4.18.0-339.el8 -> 4.18.0-340.el8)
  libdb (5.3.28-40.el8 -> 5.3.28-41.el8_4)
  libdb-utils (5.3.28-40.el8 -> 5.3.28-41.el8_4)
  libxml2 (2.9.7-11.el8 -> 2.9.7-9.el8_4.2)
  llvm-libs (12.0.1-1.module+el8.5.0+11871+08d0eab5 -> 12.0.1-2.module+el8.5.0+12488+254d2a07)
  lvm2 (8:2.03.12-6.el8 -> 8:2.03.12-8.el8)
  lvm2-libs (8:2.03.12-6.el8 -> 8:2.03.12-8.el8)
  python3-libxml2 (2.9.7-11.el8 -> 2.9.7-9.el8_4.2)
  python3-perf (4.18.0-339.el8 -> 4.18.0-340.el8)
  strace (5.7-2.el8 -> 5.7-3.el8)
```

marusak commented:
Installing it in vm.install is enough

jkozol commented:
> `rhel-8-5@osbuild/cockpit-composer` fails on `testUserAccount`. This test has been busted for a while, right @jkozol ?

Yep this is a known failure that is on the top of the fix list for me.
